perf/ssl_ctx.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <openssl/ssl.h>
+#include "perflib/perflib.h"
+
+#define NUM_CTX_PER_RUN        100000
+
+int err = 0;
+
+OSSL_TIME *times;
+
+static int threadcount;
+
+static void do_create_ctx(size_t num)
+{
+    int ret = 1;
+    int i;
+    SSL_CTX *ctx = NULL;
+    OSSL_TIME start, end;
+
+    start = ossl_time_now();
+
+    for (i = 0; i < NUM_CTX_PER_RUN / threadcount; i++) {
+        ctx = SSL_CTX_new(TLS_server_method());
```

Comment:
For a server method I think there needs to be at least some minimal configuration for it, e.g. adding a certificate and key. But perhaps we also need to do the same test for the client method - ensuring we configure a trust store etc for a client.

Reply:
Ok, I can add a command line option to select a client/server tls method, and provide options for a cert/key or keystore